packages/react-components/react-drawer/src/components/OverlayDrawer/useOverlayDrawer.ts

```diff
@@ -14,30 +14,29 @@ import { OverlayDrawerSurface } from './OverlayDrawerSurface';
  * before being passed to renderOverlayDrawer_unstable.
  *
  * @param props - props from this instance of OverlayDrawer
- * @param ref - reference to root HTMLDivElement of OverlayDrawer
+ * @param ref - reference to root HTMLElement of OverlayDrawer
  */
 export const useOverlayDrawer_unstable = (
   props: OverlayDrawerProps,
-  ref: React.Ref<HTMLDivElement>,
+  ref: React.Ref<HTMLElement>,
 ): OverlayDrawerState => {
   const { open, size, position } = useDrawerDefaultProps(props);
   const { modalType = 'modal', inertTrapFocus, onOpenChange } = props;
 
-  const motion = useMotion<HTMLDivElement>(open);
+  const motion = useMotion<HTMLElement>(open);
 
   const backdropProps = slot.resolveShorthand(props.backdrop);
   const hasCustomBackdrop = modalType !== 'non-modal' && backdropProps !== null;
 
   const root = slot.always(
-    {
+    getIntrinsicElementProps(props.as || 'div', {
```

Comment:
again, not sure this is necessary, you can just pass `div` here, `getIntrinsicElementProps` will respect the `props.as` from the spreading you're doing on the line 34